src/opm/parser/eclipse/Parser/ParserKeyword.cpp

```diff
@@ -425,8 +441,13 @@ void set_dimensions( ParserItem& item,
         if( this->m_records.empty() )
             throw std::invalid_argument( "Trying to get record from empty keyword" );
 
-        if( recordIndex >= this->m_records.size() )
-            return this->m_records.back();
+        if( recordIndex >= this->m_records.size() ) {
+            if (alternating_keyword) {
+                return this->m_records[ recordIndex % 2 ];
```

Comment:
I guess this could just be: `return this->m_records[ recordIndex % this->m_records.size() ]` - then requirement of *exactly* two records could be removed?